assets/js/googlesitekit/widgets/components/URLSearchWidget.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * URLSearchWidget component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import Widget from './Widget';
+import { STORE_NAME as CORE_SITE } from '../../datastore/site/constants';
+import Button from '../../../components/button';
+import PostSearcherAutoSuggest from '../../../components/PostSearcherAutoSuggest';
+
+const { useSelect } = Data;
+
+function URLSearchWidget() {
+	const [ canSubmit, setCanSubmit ] = useState( false );
+	const [ match, setMatch ] = useState( {} );
+
+	const detailsURL = useSelect( ( select ) => select( CORE_SITE ).getAdminURL( 'googlesitekit-dashboard', { permaLink: match?.permalink } ) );
+
+	const onClick = useCallback( () => {
+		if ( match?.permalink ) {
+			global.location.assign( detailsURL );
+		}
+	}, [ detailsURL, match ] );
```

Comment:
@caseymorrisus we can't use objects as a dependency, let's use permalink's value instead:

```suggestion
	}, [ detailsURL, match?.permalink ] );
```

Reply:
@eugene-manuilov I think it's fine to use an object as a dependency here and avoid variable assignment. Deps for hooks use referential equality comparison for objects. If we were comparing 2 completely different objects this would be an issue but as we're comparing an object to a possible future version of itself, if it is equal, the reference will also be equal. Match is also only set as a whole object meaning that any change would also modify the reference.

Reply:
Ok, sounds good then!